docs/core/tutorial/02-etl-flow.md

```diff
@@ -0,0 +1,146 @@
+- What happends to the already-transformed data if the database is not available?
+
+These points highlight the fact that `extract()` and `load()` are still arbitrarily scoped. This brings us to a rule of thumb when deciding how large to make each function: **_look at the input and output data that your workflow needs at each step._** In our case the reference data and live data come from different sources and are stored separately. Lets refactor a bit more, taking this new insight into consideration:
+
+```python
+def extract_reference_data(...):
+    # fetch reference data
+    ...
+    return reference_data
+
+def extract_live_data(...):
+    # fetch live data
+    ...
+    return live_data
+
+def transform(live_data, reference_data):
+    # clean the live data
+    ...
+    return transformed_data
+
+def load_reference_data(reference_data):
+    # save reference data to the database
+    ...
+
+def load_live_data(transformed_data):
+    # save transformed live data to the database
+    ...
+```
+
+## Leveraging Prefect
+
+Now that we have appropriately sized functions and an idea of how these functions relate to one another, let's encapsulate our workflow with Prefect.
+
+### **First step**
+
+**Decorate any function that you would like Prefect to run with `prefect.task`:**
+
+```python{1-3,9,15,21,26}
+from prefect import task, Flow
+
+@task
+def extract_reference_data(...):
+    # fetch reference data
+    ...
+    return reference_data
+
+@task
+def extract_live_data(...):
+    # fetch live data
+    ...
+    return live_data
+
+@task
+def transform(live_data, reference_data):
+    # clean the live data
+    ...
+    return transformed_data
+
+@task
+def load_reference_data(reference_data):
+    # save reference data to the database
+    ...
+
+@task
+def load_live_data(transformed_data):
+    # save transformed live data to the database
+    ...
+```
+
+### Second step
+
+**Specify data and task dependencies within a `prefect.Flow` context:**
+
+```python
+# ...task definitions above
+
+with Flow("Aircraft-ETL") as flow:
+    reference_data = extract_reference_data()
+    live_data = extract_live_data()
+
+    transformed_live_data = transform(live_data, reference_data)
+
+    load_reference_data(reference_data)
+    load_live_data(transformed_live_data)
+```
+
+Note, **this does not execute your flow**, the `with Flow(...):` context manager allows Prefect to reason about dependencies between tasks and build an execution graph that will _later_ be executed. In this case, the execution graph would look like so:
+
+![Graph ETL](/prefect-tutorial-etl-dataflow.png)
+
+A huge improvement over our original implementation!
+
+### Third step
+
+**Execute the Flow!**
+
+```python
+# ...flow definition above
+
+flow.run()
+```
+
+By this point the Tasks (your Python functions) are executed in the appropriate order, with data being passed from task-to-task as specified in the execution graph.
```

Comment:
Could we add a `:::tip Task Library` somewhere on this page to point users to our Task Library for "common operations"?  I'm going to work on a "Contributing to the Task Library" doc soon that we could also link to from this tip box.